File: app/src/main/java/com/botijasoftware/utils/ArrayBufferInt.java
package com.botijasoftware.utils;

import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import java.nio.IntBuffer;

public class ArrayBufferInt extends ArrayBuffer {
	private IntBuffer mBuffer;
	private final static int int_size = Integer.SIZE/8;
	

	public ArrayBufferInt(VertexBufferDefinition vbdefinition, int maxelements) {
		super(vbdefinition, maxelements);

		mBuffer = ByteBuffer.allocateDirect(
				vbdefinition.getDataSize() * vbdefinition.mSize * maxelements * int_size)
				.order(ByteOrder.nativeOrder()).asIntBuffer();
	}
	
	public IntBuffer getBuffer() {
		return mBuffer;
	}
	
	public void put(int index, int value) {
		mBuffer.put(index, value);
	}
	
	public void put(int[] data, int offset) {
		mBuffer.put(data, offset, data.length - offset);
		mBuffer.position(0);
	}
	
	public void put(int[] data) {
		mBuffer.put(data, 0, data.length);
		mBuffer.position(0);
	}
	
	public void put(byte value) {
		mBuffer.put(value);
	}
	
	public void put(short value) {
		mBuffer.put(value);
	}
	
	public void put(int value) {
		mBuffer.put(value);
	}
	
	public void put(float value) {
		mBuffer.put((int)value);
	}
	
	public void position(int i) {
		mBuffer.position(i);
	}
	
	public int position() {
		return mBuffer.position();
	}

}

File: app/src/main/java/com/botijasoftware/utils/ArrayBufferInt.kt
package com.botijasoftware.utils

import java.nio.ByteBuffer
import java.nio.ByteOrder
import java.nio.IntBuffer

class ArrayBufferInt(vbdefinition: VertexBufferDefinition, maxelements: Int) : ArrayBuffer(vbdefinition, maxelements) {
    override val buffer: IntBuffer


    init {

        buffer = ByteBuffer.allocateDirect(
                vbdefinition.dataSize * vbdefinition.mSize * maxelements * int_size)
                .order(ByteOrder.nativeOrder()).asIntBuffer()
    }

    override fun put(index: Int, value: Int) {
        buffer.put(index, value)
    }

    override fun put(data: IntArray, offset: Int) {
        buffer.put(data, offset, data.size - offset)
        buffer.position(0)
    }

    override fun put(data: IntArray) {
        buffer.put(data, 0, data.size)
        buffer.position(0)
    }

    override fun put(value: Byte) {
        buffer.put(value.toInt())
    }

    override fun put(value: Short) {
        buffer.put(value.toInt())
    }

    override fun put(value: Int) {
        buffer.put(value)
    }

    override fun put(value: Float) {
        buffer.put(value.toInt())
    }

    override fun position(i: Int) {
        buffer.position(i)
    }

    override fun position(): Int {
        return buffer.position()
    }

    companion object {
        private val int_size = Integer.SIZE / 8
    }

}